### ISSUE TITLE
Tune not checkpointing at end for all trials

### ISSUE THREAD

**Issue by waldroje** [ray-project/ray#5745]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: RHEL 7.6
- **Ray installed from (source or binary)**: Source
- **Ray version**: 0.7.4
- **Python version**: 3.6.8
- **Exact command to reproduce**:

### Describe the problem

I"m getting inconsistent behavior from checkpoint uploads... on longer running trials, often I'm not getting the final checkpoint, particularly when it's run with a "grid-search", as it seems whatever trial is the last trial running fails to get the final checkpoint uploaded.

Doing something simple as below, I'm also not getting an uploaded checkpoint... even though I do see the checkpoint in my local directory... all the other files are being uploaded, just not checkpoints. Possibly it's just me... but wondering if you get the same behavior... my upload_dir is a simple "s3://bucket/ray_results" type path, and as I indicated, it's not that it doesn't ever work... so I know there is not an issue with the path.

### Source code / logs

```
run("PPO",
    name="checkpoint-test",
    **{
        "checkpoint_at_end": True,
        "upload_dir": UPLOAD_DIR,
        "stop": {"episodes_total": 2000},
        "config": {
            "num_workers": 2,
            "num_gpus": 1.0,
            "env": "CartPole-v0"
        },
        "verbose": 1
    }
)
```

**Comment by richardliaw** [comment]
Should be fixed in #5699; can you try out master?

**Comment by waldroje** [comment]
Sorry, I did a quick search and nothing popped... didn't think that it was already fixed... thanks much
